bundles/org.openhab.core/src/main/java/org/openhab/core/common/registry/ManagedProvider.java

```diff
@@ -63,6 +65,6 @@
      * @return returns element or null, if no element for the given key exists
      */
     @Nullable
-    E get(K key);
+    E get(@NonNull K key);
```

Comment:
It seems that `@NonNullByDefault` is not applied to generic type parameters. So it would be nice if SAT will not generate warnings for these.

Reply:
I tried adding `@NonNull` to some generic type parameters but it doesn't seem to be used at all this way. :neutral_face:

Reply:
Have you tried to state that K in general extends a non null object?

Reply:
It seems to work and you can also override the annotations at method level. But it has a domino effect so I have to add/fix more annotations to get this working. :wink:

Reply:
Apparently Xtend doesn't support type annotations 

https://bugs.eclipse.org/bugs/show_bug.cgi?id=506374

So there is no way to satisfy the nullness annotations of the `GenericThingProvider` causing the compiler to error on this project:

https://github.com/openhab/openhab-core/blob/aae207bfe95da3b0b7eaaf08e4beda546a1c228d/bundles/org.openhab.core.model.thing/src/org/openhab/core/model/thing/internal/GenericThingProvider.xtend#L76

For now I've reconfigured the compiler to only log warnings for such issues on this project.